chapter09/pxd/ex2.pl

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use feature "say";
+
+open my $fh, '<', $ARGV[0]?$ARGV[0]:'input.txt'
+    or die "Can't open input.txt: $!";
+
+my @patterns;
+
+while (<$fh>) {
+    chomp;
+    my $pattern = eval { qr/$_/ }; 
```

Comment:
Unchecked eval is a bad habit, it's usually worth doing something with ``$@``, even if it's just a warning.